examples/dataflow-streaming-benchmark/src/main/java/com/google/cloud/pso/pipeline/StreamingBenchmark.java

```diff
@@ -180,8 +202,52 @@ public static PipelineResult run(Options options) {
       this.schemaLocation = schemaLocation;
     }
 
+    // Leave the scope as package private since its referred inside test methods for assertions
+    List<String> getAttributeFields() {
+      return this.attributeFields;
+    }
+
+    /**
+     * Validate Schema and checks for presence of attribute schema fields
+     *
+     * @param dataGenerator The execution options.
+     * @param schema The execution options.
+     * @throws IOException, MalformedSchemaException
+     */
+    private void validateSchema(JsonDataGenerator dataGenerator, String schema)
+        throws IOException, MalformedSchemaException {
+
+      String payload;
+      JsonObject jsonObject;
+
+      try {
+        // Generate sample message based on the provided schema.
+        try (ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream()) {
+          dataGenerator.generateTestDataJson(schema, byteArrayOutputStream);
+          payload = byteArrayOutputStream.toString();
+        }
```

Comment:
Can we encapsulate this code in another function given that it is repeated in the `processElement` method?

Reply:
Initially I thought the same but intentionally avoided (given that repeated code was only 2 lines) to reduce the number of function calls that needs to be made inside process element  during high volume stream of messages.

Reply:
Sure but since this is running on the JVM the JIT compiler should inline it given it's frequency & size so the method call becomes close to free.